File: tests/libjournal/cout.cc
// -*- c++ -*-
//
// michael a.g. aïvázis <michael.aivazis@para-sim.com>
// (c) 1998-2020 all rights reserved


// get the journal
#include <j2/journal.h>
// support
#include <cassert>


// alias the type
using console_t = pyre::journal::cout_t;


// exercise the trivial device
int main() {
    // instantiate
    console_t console;
    // check its name
    assert (console.name() == "cout");

    // all done
    return 0;
}


// end of file

File: tests/journal.api/cout.py
#! /usr/bin/env python3
# -*- coding: utf-8 -*-
#
# michael a.g. aïvázis <michael.aivazis@para-sim.com>
# (c) 1998-2020 all rights reserved


def test():
    """
    Verify that the console is accessible
    """
    # access
    import j2

    # instantiate the console
    console = j2.cout()
    # verify its name is what we expect
    assert console.name == "cout"

    # all done
    return


# main
if __name__ == "__main__":
    # run the test
    test()


# end of file
